cmd/cmd.go

```diff
@@ -1,17 +1,152 @@
 package cmd
 
 import (
+	"encoding/json"
 	"fmt"
 
 	"github.com/spf13/cobra"
+
+	"github.com/ignite/cli-plugin-airdrop/pkg/config"
+	"github.com/ignite/cli-plugin-airdrop/pkg/genesis"
+	"github.com/ignite/cli-plugin-airdrop/pkg/snapshot"
 )
 
 func NewAirdrop() *cobra.Command {
-	return &cobra.Command{
+	c := &cobra.Command{
 		Use:   "airdrop",
 		Short: "Utility tool to create snapshots for an airdrop",
+	}
+
+	c.AddCommand(
+		NewAirdropGenerate(),
+		NewAirdropRaw(),
+		NewAirdropProcess(),
+		NewAirdropGenesis(),
+	)
+
+	return c
+}
+
+func NewAirdropGenerate() *cobra.Command {
+	return &cobra.Command{
+		Use:   "generate [airdrop-config] [input-genesis]",
+		Short: "Utility tool to create snapshots for an airdrop",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				airdropConfig = args[0]
+				inputGenesis  = args[1]
+			)
+
+			c, err := config.ParseConfig(airdropConfig)
+			if err != nil {
+				return err
+			}
+
+			genState, err := genesis.GetGenStateFromPath(inputGenesis)
+			if err != nil {
+				return err
+			}
+
+			s, err := snapshot.Generate(genState)
+			if err != nil {
+				return err
+			}
+
+			filters := make(snapshot.Filters, 0)
+			for _, snap := range c.Snapshots {
+				filter := s.Filter(snapshot.FilterType(snap.Type), snap.Denom, snap.Formula, snap.Excluded)
+				filters = append(filters, filter)
+			}
+			filter := filters.Sum()
+
+			// export snapshot json
+			filterJSON, err := json.MarshalIndent(filter, "", "    ")
+			if err != nil {
+				return fmt.Errorf("failed to marshal snapshot: %w", err)
+			}
+
+			cmd.Println(string(filterJSON))
+			// TODO generate the genesis
+			return nil
+		},
+	}
+}
+
+func NewAirdropRaw() *cobra.Command {
+	return &cobra.Command{
+		Use:   "raw [input-genesis]",
+		Short: "Generate raw airdrop data based on the input genesis",
```

Comment:
What is the input genesis for the raw command? Is it the genesis that has been exported from the chain to parse snapshot from? It's a bit unclear since we should also have the "input genesis" where we populate the claim genesis state from the snapshot data

This may be implemented but I think we should let the use the ability to provide a RPC address for the chain, where the genesis would be automatically exported